Apps/phrequesttracker/requesttracker_connector.py

```diff
@@ -0,0 +1,725 @@
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _get_ticket_details(self, ticket_id, action_result):
+
+        # Query the device for details about the ticket
+        ret_val, resp_text = self._make_rest_call("ticket/{0}/show".format(ticket_id), action_result)
+
+        if phantom.is_fail(ret_val):
+            self.save_progress("Get ticket details for id '{0}' failed")
+            return ret_val
+
+        ticket_info = {}
+
+        self.debug_print(resp_text)
+
+        for line in resp_text.split('\n'):
+            if ':' in line:
+                spl_line = line.split(':')
+                key = spl_line[0]
+                value = ':'.join(spl_line[1:])
+                ticket_info[key] = value.strip()
+
+        data = {}
+        data[RT_JSON_ID] = ticket_id
+        data[RT_JSON_QUEUE] = ticket_info.get('Queue')
+        data[RT_JSON_OWNER] = ticket_info.get('Owner')
+        data[RT_JSON_CREATOR] = ticket_info.get('Creator')
+        data[RT_JSON_SUBJECT] = ticket_info.get('Subject')
+        data[RT_JSON_STATUS] = ticket_info.get('Status')
+        data[RT_JSON_PRIORITY] = ticket_info.get('Priority')
+        data[RT_JSON_INITIALPRIORITY] = ticket_info.get('InitialPriority')
+        data[RT_JSON_FINALPRIORITY] = ticket_info.get('FinalPriority')
+        data[RT_JSON_REQUESTORS] = ticket_info.get('Requestors')
+        data[RT_JSON_CC] = ticket_info.get('Cc')
+        data[RT_JSON_ADMINCC] = ticket_info.get('AdminCc')
+        data[RT_JSON_CREATED] = ticket_info.get('Created')
+        data[RT_JSON_STARTS] = ticket_info.get('Starts')
+        data[RT_JSON_STARTED] = ticket_info.get('Started')
+        data[RT_JSON_DUE] = ticket_info.get('Due')
+        data[RT_JSON_RESOLVED] = ticket_info.get('Resolved')
+        data[RT_JSON_TOLD] = ticket_info.get('Told')
+        data[RT_JSON_TIMEESTIMATED] = ticket_info.get('TimeEstimated')
+        data[RT_JSON_TIMEWORKED] = ticket_info.get('TimeWorked')
+        data[RT_JSON_TIMELEFT] = ticket_info.get('TimeLeft')
+
+        action_result.add_data(data)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _get_ticket(self, param):
+
+        # Create action results
+        action_result = self.add_action_result(ActionResult(param))
+
+        # Create RT session
+        if phantom.is_fail(self._create_rt_session(action_result)):
+            return action_result.get_status()
+
+        # get the ticket ID
+        ticket_id = param[RT_JSON_ID]
+
+        return self._get_ticket_details(ticket_id, action_result)
+
+    def _list_tickets(self, param):
+
+        # Create the action result
+        action_result = self.add_action_result(ActionResult(param))
+
+        # Create RT session
+        if phantom.is_fail(self._create_rt_session(action_result)):
+            return action_result.get_status()
+
+        # Set up the query
+        query = "Queue='{0}' AND {1}".format(param.get(RT_JSON_QUEUE, DEFAULT_QUEUE), param.get(RT_JSON_QUERY, ''))
```

Comment:
Just want to make sure that if you get `''` for the second param, the query still works

Reply:
Reworked.